numba/cuda/cudadrv/driver.py

```diff
@@ -221,9 +221,12 @@ def __init__(self):
             self.lib = find_driver()
         except CudaSupportError as e:
             self.is_initialized = True
-            self.initialization_error = e
+            self.initialization_error = e.msg
+
+    def ensure_initialized(self):
```

Comment:
As noted out of band, am not convinced this (or the init) is threadsafe in its current state, seems like the `is_initialized` bit is set too eagerly. Suggest fixing for the next release. xref https://github.com/numba/numba/issues/7387